### DISCUSSION TITLE
ModuleNotFoundError when importing telegram module with Python-Telegram-Bot

### DISCUSSION THREAD

**Discussion by consci210** [python-telegram-bot/python-telegram-bot#3605] [Q&A]
I am encountering an error when trying to use the telegram bot with Python. Specifically, when importing the necessary modules with the following code:

**`import logging
from telegram.ext import Updater, CommandHandler, MessageHandler, Filters
`**
I consistently receive the following error message:

**`ModuleNotFoundError: No module named 'telegram'`**

I have tried several solutions including:

- Running the commands `pip install python-telegram-bot`,` pip3 install python-telegram-bot`, `pip install telegram`, and `pip3 install telegram`

- Uninstalling and reinstalling the modules separately

- Renaming the file to avoid any module import conflicts

- Running the commands locally on the virtual environment as well as from the cmd terminal

Despite all these efforts, I have not been able to resolve the issue.If anyone could provide assistance or guidance, I would greatly appreciate it

**Comment by Poolitzer** [comment]
Hey there. You need to install `python-telegram-bot`, not `telegram-bot`. Check out this FAQ link if you still have issues. [Why am I getting ImportError: cannot import name 'XY' from 'telegram'?](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Frequently-Asked-Questions#why-am-i-getting-importerror-cannot-import-name-xy-from-telegram)

**Reply by consci210** [reply]
Heyyy...Thank you for your response. I appreciate your suggestion to install python-telegram-bot instead of telegram-bot. However, I have tried installing python-telegram-bot but unfortunately, the issue still persists.

**Reply by Poolitzer** [reply]
if the pip command runs through without an error and you still get ModuleNotFound, something is fundamentally wrong on your system (or you are calling the wrong env). Helping with this is not in our scope, you would need to ask the community behind pip I guess.
